src/engine/shared/datafile.h

```diff
@@ -114,7 +114,7 @@ class CDataFileWriter
 	};
 
 	IOHANDLE m_File;
-	std::array<CItemTypeInfo, MAX_ITEM_TYPES> m_aItemTypes;
+	std::map<int, CItemTypeInfo, std::less<>> m_ItemTypes;
```

Comment:
```suggestion
	std::map<uint16_t, CItemTypeInfo> m_ItemTypes;
```
`std::less<>` is default I think. Item types can only be `uint16_t`s.

Reply:
I replaced it with `uint16_t`. I prefer the explicit `std::less<>` though, as the item types must be sorted in ascending order (or at least that was previously the assumption with datafiles)